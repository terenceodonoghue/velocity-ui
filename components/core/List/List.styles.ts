import { css } from '@emotion/react';

const listRoot = css`
  margin: 0;
  padding: 0;
  list-style: none;
`;

export default listRoot;
