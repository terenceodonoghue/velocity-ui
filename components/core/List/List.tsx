import React, { FunctionComponent } from 'react';
import styled from '@emotion/styled';
import flexRoot from './List.styles';
import type { ListProps } from './List.types';

const ListRoot = styled.ul(flexRoot);

const List: FunctionComponent<ListProps> = ({ children, component = 'ul' }) => (
  <ListRoot as={component} data-testid={`list-${component}`}>
    {children}
  </ListRoot>
);

export default List;
