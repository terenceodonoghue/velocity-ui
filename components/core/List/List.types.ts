export interface ListProps {
  component?: 'ol' | 'ul';
}
