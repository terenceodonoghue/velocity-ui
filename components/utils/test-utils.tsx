import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
