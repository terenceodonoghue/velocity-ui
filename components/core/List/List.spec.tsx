import React from 'react';
import { render } from 'test-utils';
import List from './List';

describe('List', () => {
  it('is unordered by default', () => {
    const { container, getByTestId } = render(<List />);

    expect(getByTestId('list-ul')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders ordered list', () => {
    const { container, getByTestId } = render(<List component="ol" />);

    expect(getByTestId('list-ol')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders unordered list', () => {
    const { container, getByTestId } = render(<List component="ul" />);

    expect(getByTestId('list-ul')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
