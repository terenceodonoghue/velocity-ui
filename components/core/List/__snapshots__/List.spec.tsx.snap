// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List is unordered by default 1`] = `
.emotion-0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    data-testid="list-ul"
  />
</div>
`;

exports[`List renders ordered list 1`] = `
.emotion-0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

<div>
  <ol
    class="emotion-0 emotion-1"
    data-testid="list-ol"
  />
</div>
`;

exports[`List renders unordered list 1`] = `
.emotion-0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    data-testid="list-ul"
  />
</div>
`;
